megatron/fused_kernels/__init__.py

```diff
@@ -31,62 +32,87 @@ def load(args):
 
     # Check if cuda 11 is installed for compute capability 8.0
     cc_flag = []
-    _, bare_metal_major, _ = _get_cuda_bare_metal_version(
-        cpp_extension.CUDA_HOME)
-    if int(bare_metal_major) >= 11:
-        cc_flag.append('-gencode')
-        cc_flag.append('arch=compute_80,code=sm_80')
+    if torch.version.hip is None:
+        _, bare_metal_major, _ = _get_cuda_bare_metal_version(
+            cpp_extension.CUDA_HOME)
+        if int(bare_metal_major) >= 11:
+            cc_flag.append('-gencode')
+            cc_flag.append('arch=compute_80,code=sm_80')
 
     # Build path
     srcpath = pathlib.Path(__file__).parent.absolute()
     buildpath = srcpath / 'build'
     _create_build_dir(buildpath)
 
     # Helper function to build the kernels.
-    def _cpp_extention_load_helper(name, sources, extra_cuda_flags):
-        return cpp_extension.load(
-            name=name,
-            sources=sources,
-            build_directory=buildpath,
-            extra_cflags=['-O3',],
-            extra_cuda_cflags=['-O3',
-                               '-gencode', 'arch=compute_70,code=sm_70',
-                               '--use_fast_math'] + extra_cuda_flags + cc_flag,
-            verbose=(args.rank == 0)
-        )
+    def _cpp_extention_load_helper(name, sources, extra_cuda_flags, extra_include_paths):
+        if torch.version.hip is not None:
```

Comment:
I feel this code could be simplified so that the only branch is on the value of `extra_cuda_cflags`.

Reply:
I made the suggested change.

cc: @jithunnair-amd

Reply:
@rraminen, I apologize for my vague feedback. This is not what I meant. 

What I meant was that the 2 return commands only seem to differ in the value of `extra_cuda_cflags`. So, I suggested having just one return like
```python
    if torch.version.hip is not None:
       extra_cuda_cflags = ...
    else:
       extra_cuda_cflags = ...

    return cpp_extension.load(
                ...
      )

Reply:
Sorry, I made the changes.